### PR TITLE
PA-2571 do not allow review form to go into view only mode if that se…

### PR DESCRIPTION
**Pull request by devinleighsmith**
…ction has errors.

Also add support for popper props within the datepicker.

### CODE REVIEW

File: frontend/src/features/mapSideBar/SidebarContents/subforms/AssociatedLandReviewPage.tsx [hunk 1]
```diff
@@ -132,7 +132,7 @@ export const AssociatedLandReviewPage: React.FC<any> = (props: IReviewProps) =>
                         className="edit"
                         onClick={() =>
                           setEditInfo({
-                            ...defaultEditValues,
```
Comment (devinleighsmith) [inline frontend/src/features/mapSideBar/SidebarContents/subforms/AssociatedLandReviewPage.tsx:135]: This was causing the other sections to change when an unrelated edit icon was clicked. I think I like it better if only the clicked section changes.

File: frontend/src/features/mapSideBar/SidebarContents/subforms/BuildingReviewPage.tsx [hunk 2]
```diff
@@ -265,6 +268,13 @@ export const BuildingReviewPage: React.FC<any> = (props: IReviewProps) => {
                     formikProps={formikProps}
                     disabled={editInfo.tenancy}
                     field={withNameSpace('buildingTenancyUpdatedOn')}
+                    popperModifiers={{
```
Comment (devinleighsmith) [inline frontend/src/features/mapSideBar/SidebarContents/subforms/BuildingReviewPage.tsx:271]: This prevents the popup from being clipped by the parent component.

File: frontend/src/features/mapSideBar/SidebarContents/subforms/BuildingReviewPage.tsx [hunk 3]
```diff
@@ -76,8 +76,8 @@ export const BuildingReviewPage: React.FC<any> = (props: IReviewProps) => {
                     className="edit"
                     onClick={() =>
                       setEditInfo({
-                        ...defaultEditValues,
-                        identification: !editInfo.identification,
+                        ...editInfo,
+                        identification: formikProps.isValid && !editInfo.identification,
```
Comment (devinleighsmith) [inline frontend/src/features/mapSideBar/SidebarContents/subforms/BuildingReviewPage.tsx:80]: Do not allow the section to be placed in read only mode if there are validation errors as those errors are only displayed in edit mode.

File: frontend/src/components/common/form/FastDatePicker.tsx [hunk 4]
```diff
@@ -7,6 +7,7 @@ import { formikFieldMemo } from 'utils';
 import classNames from 'classnames';
 import GenericModal from '../GenericModal';
 import { appraisalDateWarning } from 'features/projects/common';
+import * as Popper from 'popper.js';
```
Comment (agalbrai) [inline frontend/src/components/common/form/FastDatePicker.tsx:10]: googled this to see what it was, that's super handy! also the website is cute lol

**Review by agalbrai — APPROVED** [review]

**Review by Fosol — APPROVED** [review]
